dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -378,38 +379,64 @@ public ResourceSupport findRel(HttpServletRequest request, HttpServletResponse r
     /**
      * Execute a POST request;
      *
-     * curl -X POST http://<dspace.url>/dspace-spring-rest/api/{apiCategory}/{model}
+     * curl -X POST -H "Content-Type:application/json" http://<dspace.url>/dspace-spring-rest/api/{apiCategory}/{model}
      *
      * Example:
      * <pre>
      * {@code
-     *      curl -X POST http://<dspace.url>/dspace-spring-rest/api/submission/workspaceitems
+     *      curl -X POST -H "Content-Type:application/json" http://<dspace.url>/dspace-spring-rest/api/submission/workspaceitems
      * }
      * </pre>
      *
-     * @param request
-     * @param apiCategory
-     * @param model
-     * @return
-     * @throws HttpRequestMethodNotSupportedException
+     * @param request       The relevant request
+     * @param apiCategory   The apiCategory to be used
+     * @param model         The model to be used
+     * @return              The relevant ResponseEntity for this request
+     * @throws HttpRequestMethodNotSupportedException   If something goes wrong
      */
-    @RequestMapping(method = RequestMethod.POST)
+    @RequestMapping(method = RequestMethod.POST, consumes = {"application/json", "application/hal+json"})
```

Comment:
I'm confused as to why we now have two `POST` options, one for JSON and one for "text/uri-list" (see next method).  I thought we had decided to only support "text/uri-list" in our discussion in this PR.

To me, it seems like we should drop this method and `postJsonInternal()`, and rename the "text/uri-list" version to be `post()`.

Reply:
This is at the `RestResourceController` level, where `POST` can occur with JSON or with a uri-list.
The `@RequestMapping(method = RequestMethod.POST, consumes = {"application/json", "application/hal+json"})` will be used for e.g. `CommunityRestRepository.createAndReturn`.
The `@RequestMapping(method = RequestMethod.POST, consumes = {"text/uri-list"})` will be used for e.g. `RelationshipRestRepository.createAndReturn`

So yes, we only support "text/uri-list" in the RelationshipRestRepository, but the implementation also had to make sure the JSON functionality is still supported and the solution is reusable for the other features requiring "text/uri-list"

Reply:
@benbosman : Thanks for pointing that out! I completely missed the fact that this was a "parent" Controller that is used from multiple other Controllers.  Disregard this comment, as your explanation makes perfect sense.  I'll give this a re-review.